photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -138,9 +156,9 @@ class CoordinateDescent(
               .setName(s"Updated model with coordinateId $coordinateId at iteration $iteration")
               .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
               .materialize()
-
           case _ =>
         }
+        // We need to split out the following 2 match expressions: [[FactoredRandomEffectModel]] matches both
         oldModel match {
           case broadcastLike: BroadcastLike => broadcastLike.unpersistBroadcast()
           case _ =>
```

Comment:
@ferd36 The fixed effect model is Broadcast-like because it's a single model that gets distributed to the executors to be aggregated in parallel. Random effect models are RDD-like because each random-effect problem is solved individually on a single executor, so we map over them like an RDD.